emulate-browsers/base/data-scripts/updateScreenSizesStats.ts

```diff
@@ -1,52 +0,0 @@
-import moment from 'moment';
```

Comment:
I agree we don't want to auto-rotate these, but I think we might want to keep the ability to periodically import the top screen sizes. I don't know where exactly we want this to live if it's turned off by default though... ie, how does a user get random sizes?

Reply:
Maybe we need a separate package for randomizing values - like browsers, screensizes, timezone+languages, etc. If we're not going do turn these on by default, it might be nice to have a "randomizer" package to help do so.

Reply:
The more I've thought about this the more I believe that it's highly unlikely a developer will ever want to randomize screen sizes. It's because of MediaQueries. They affect whether an element is even in the DOM, which seems to make randomization dangerous.

Also, how often do you think the top screen size changes? It's years, right?

Reply:
I still think you're thinking about this edge-case in. Yes, it is possible that you might hide an element with a media query, but I think if you are rotating through common "desktop" screen sizes, the range that will impact an element you want to scrape is wider than the randomization. Ie, you will mostly see no changes to elements of importance rotating through desktop screen sizes. I don't have data for that, just going off of my experience using media queries. You hide elements that are things like nav, not the core data.

Separately, screen size is part of fingerprinting. I don't see how we think we can send across the same screen size, position, and browser window size every time and think that won't be detected.

Reply:
Stats, fwiw: https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide/#monthly-201910-202010

Reply:
Anything that stood out to you in the stats?

Reply:
Screen size is a very minor piece of fingerprinting. Look at the stat percentages. The bucket that your screen size places you in is vast. The screen size on my 15" MacBook Pro is the same size as almost everyone else who has the 15" Macbook Pro.

Reply:
And... I don't know what "mostly" means when you say, "you will mostly see no changes to elements of importance rotating through desktop screen sizes"... if you have a scraper script that hinges something off of a navigation tab and the rotating screens means that your script breaks 10% of time... that's going to be a huge hit to your scraping project, and it's going to be reaaaaaaaaly frustrating to debug... and you're going to blame the issue on us, which you should.

Reply:
I guess you must see this very differently than I do. I don't think your script will break 10% of the time if you chose to implement a screen rotation strategy when using the top 4 resolutions on that list. My uninformed guess would be 0% for 95% of scripts. I guess I just don't think the web looks like a place where media queries are impacting desktop resolutions that drastically.

If we include mobile and/or orientation differences, I agree with your sentiment.

Two things stood out to me about the stats:
1) The top screen size represents 20% of traffic. We made 50% of traffic our goal for browsers, so it seemed like we might want to do the same here
2) The top 4 have a bigger width jump than height, but unless there's different settings for >1366 width and >766 width, I think this is likely the same bucket of media query resolutions. Hard to imagine the use cases where designers are selectively changing things for this large setup.

Reply:
Let's discuss further on the phone... I agree with your guess of "0% for 95% of scripts". It's the 5% that bothers me because the 5% turns into a massive frustration that's incredibly difficult to debug for those 5%.

Reply:
An idea... 

I could see setting different screen sizes, but picking a browser width/height that can be consistent within those screen sizes..

Reply:
It seems as of a few days ago we dropped our near-term target of 50% for browsers.... now it's just the single most popular browser.

Reply:
Another idea.. perhaps this is a finding mission-impossible intends to find out the answer to - the variety of media queries used in the wild